numba/targets/listobj.py

```diff
@@ -465,6 +465,12 @@ def list_impl(iterable):
 
     return context.compile_internal(builder, list_impl, sig, args)
 
+@lower_builtin(list)
+def list_constructor(context, builder, sig, args):
```

Comment:
I'm wonder if this gets simpler implemented as:

```python
@overload(list)
def list_impl():
     ...
     def codegen():
            return []
     return codegen
```

Reply:
Will that work? That is a much better idea if so. I went via the low level methods as I know typing empty lists is hard work from prior experience :)

Reply:
Just using the overload with no extra typing behaviour does:
`Invalid usage of Function(<class 'list'>) with parameters ()`
and I can't find a way to get the type info from inside the typing scope of the overload, IIRC it's not trivial?